src/components/Hero/HeroSearchBar/HeroSearchBar.jsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { route } from '@/constants';
+import { AppSearchInput, AppSelect } from '@/shared';
+import {
+  HeroSearchBarFormStyled,
+  HeroSearchButtonStyles,
+} from './HeroSearchBar.styled';
+
+const searchType = Object.freeze({
+  CHEF: 'Chef',
+  DISH: 'Dish',
+});
+
+const HeroSearchBar = () => {
+  const navigate = useNavigate();
+  const [type, setType] = useState(searchType.DISH);
+  const [query, setQuery] = useState('');
+
+  const changeTypeHandler = useCallback((evt) => {
+    setType(evt.target.value);
+  }, []);
+  const changeQueryHandler = useCallback((evt) => {
+    setQuery(evt.target.value);
+  }, []);
+  const submitHandler = useCallback(
+    (evt) => {
+      evt.preventDefault();
+      const searchQuery = query.trim();
+
+      navigate(
+        `${
+          type === searchType.CHEF ? route.CHEFS : route.DISHES
+        }?search=${searchQuery}`
```

Comment:
If a user doesn't enter any search term, we redirect them to a route like `http://localhost:3000/dishes?search=`. Probably it could be a good approach to add something like the word `all` in this case to make the route appear more understandable, like `http://localhost:3000/dishes?search=all`. What do you think about it?

Reply:
You definitely don't need to write ‘all’. An empty line shows us that all values should be returned without filtering. Maybe, if search is empty not need to add ?search=.